Core/Resource/Model.cs

```diff
@@ -495,49 +532,52 @@ where type.IsSubclassOf(typeof(Instance))
 
         private readonly Regex _idFromNamespace = new Regex(@"(\{){0,1}[0-9a-fA-F]{8}_[0-9a-fA-F]{4}_[0-9a-fA-F]{4}_[0-9a-fA-F]{4}_[0-9a-fA-F]{12}(\}){0,1}",
                                                             RegexOptions.IgnoreCase);
-
-        private Symbol ReadSymbolFromFile(string filepath)
+        
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static string ReplaceCharacter(string idString, char toReplace, char toReplaceWith)
         {
-            Log.Info($" read symbol: `{filepath}`");
-            using var streamReader = new StreamReader(filepath);
-            using var jsonReader = new JsonTextReader(streamReader);
+            var idReadOnlySpan = idString.AsSpan();
+            Span<char> namespaceIdSpan = stackalloc char[idReadOnlySpan.Length];
 
-            try 
+            for (int i = 0; i < idReadOnlySpan.Length; i++)
             {
-                var symbolJson = new SymbolJson { Reader = jsonReader };
-                var symbol = symbolJson.ReadSymbol(this);
-                if (symbol == null)
-                    return null;
-
-                //symbol.SourcePath = OperatorTypesFolder + symbol.Name + SourceExtension;
-                SymbolRegistry.Entries.Add(symbol.Id, symbol);
-
-                return symbol;
+                var c = idReadOnlySpan[i];
+                namespaceIdSpan[i] = c == toReplace ? toReplaceWith : c;
             }
-            catch (Newtonsoft.Json.JsonReaderException e)
+
+            return new string(namespaceIdSpan);
+        }
+        
+        internal bool TryReadSymbolWithId(Guid symbolId, out Symbol symbol)
```

Comment:
A re-do on the `ReadSymbolWithId function using the `JsonResult`s that we already got earlier